app/helpers/evaluations_helper.rb

```diff
@@ -25,19 +25,20 @@ def display_score(assignment)
 
   # individual evaluator score
   def evaluator_score(assignment)
-    score = assignment.evaluation&.total_score
-
-    unless assignment.evaluation_status == :completed && score
-      return Score.new(0, "0", "N/A")
-    end
+    score = display_score(assignment)
+    return Score.new(0, "0", "N/A") if score == 'N/A'
 
     Score.new(score, score.to_s, score)
   end
 
   def average_score(submission)
+    assigned_evaluations = submission.evaluator_submission_assignments.assigned
+
+    return Score.new(0, "0", "N/A") if assigned_evaluations.empty?
+
     completed_evaluations = submission.evaluations.where.not(completed_at: nil)
 
-    unless completed_evaluations.any?
+    unless completed_evaluations.count == assigned_evaluations.count
```

Comment:
I think there is an issue because `completed_evaluations` includes all evaluations with `completed_at: nil` however we only want to calculate and return the average_score for the completed_evaluations that are *also* `assigned` status? you can filter the query
```suggestion
    completed_evaluations = submission.evaluations.where(evaluator_submission_assignment: assigned_evaluations).where.not(completed_at: nil)
    
    unless completed_evaluations.count == assigned_evaluations.count
```

Reply:
Ahh yep, that makes sense - I'll push that change now, thanks!